tools/regex_linter.py

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env python
+import argparse
+import collections
+import re
+
+from clang_tidy import get_changed_files, get_changed_lines
```

Comment:
I don't think we should run the regex linter on changed lines only. We should run it on all lines. Otherwise you'll get in a situation like clang-tidy where people check in a breakage and we stop seeing it in contbuild.